tests/components/bond/test_fan.py

```diff
@@ -31,12 +32,60 @@ def ceiling_fan(name: str):
     }
 
 
+async def turn_fan_on(hass: core.HomeAssistant, fan_id: str, speed: str) -> None:
+    """Turn the fan on at the specified speed."""
+    await hass.services.async_call(
+        FAN_DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: fan_id, fan.ATTR_SPEED: speed},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+
 async def test_entity_registry(hass: core.HomeAssistant):
     """Tests that the devices are registered in the entity registry."""
     await setup_platform(hass, FAN_DOMAIN, ceiling_fan("name-1"))
 
     registry: EntityRegistry = await hass.helpers.entity_registry.async_get_registry()
-    assert [key for key in registry.entities.keys()] == ["fan.name_1"]
+    assert [key for key in registry.entities] == ["fan.name_1"]
+
+
+async def test_entity_non_standard_speed_list(hass: core.HomeAssistant):
+    """Tests that the device is registered with custom speed list if number of supported speeds differs form 3."""
+    await setup_platform(
+        hass,
+        FAN_DOMAIN,
+        ceiling_fan("name-1"),
+        bond_device_id="test-device-id",
+        props={"max_speed": 6},
+    )
+
+    actual_speeds = hass.states.get("fan.name_1").attributes[ATTR_SPEED_LIST]
+    assert actual_speeds == [
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5844602)